File: web/src/test/java/me/robinvdb/runetracker/database/DatabaseServiceSmokeTest.java
package me.robinvdb.runetracker.database;


import me.robinvdb.runetracker.domain.RunescapeUser;
import org.junit.Test;


import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;

public class DatabaseServiceSmokeTest {
    @Test
    public void smokeTest() {
        RunescapeUserDatabase runescapeUserDatabase = new RunescapeUserDatabaseStub();
        DatabaseService service = new DatabaseService(runescapeUserDatabase, mock(DataPointDatabase.class));

        assertThat(service.getAllUsers())
                .hasSize(0);

        RunescapeUser user = new RunescapeUser();
        service.addUser(user);

        assertThat(service.getAllUsers())
                .hasSize(1)
                .contains(user);

        service.deleteUser(user);

        assertThat(service.getAllUsers())
                .hasSize(0)
                .doesNotContain(user);
    }
}

File: web/src/test/java/me/robinvdb/runetracker/database/DatabaseServiceSmokeTest.kt
package me.robinvdb.runetracker.database


import me.robinvdb.runetracker.domain.RunescapeUser
import org.junit.Test


import org.assertj.core.api.Assertions.assertThat
import org.mockito.Mockito.mock
import org.mockito.Mockito.`when`

class DatabaseServiceSmokeTest {
    @Test
    fun smokeTest() {
        val runescapeUserDatabase = RunescapeUserDatabaseStub()
        val service = DatabaseService(runescapeUserDatabase, mock(DataPointDatabase::class.java))

        assertThat(service.allUsers)
                .hasSize(0)

        val user = mock(RunescapeUser::class.java)
        `when`<Int>(user.id).thenReturn(0)

        service.addUser(user)

        assertThat(service.allUsers)
                .hasSize(1)
                .contains(user)

        service.deleteUser(user)

        assertThat(service.allUsers)
                .hasSize(0)
                .doesNotContain(user)
    }
}
